### PR TITLE
DOC-4303 updated Go landing page

### PR DESCRIPTION
**Pull request by andy-stark-redis**
I've edited the text of the page and added some JSON and hash search examples, so it now looks more like the equivalent pages for redis-py, etc. Vector search was suggested as a possible extra example but we don't currently have any vector search code samples and none of the other client "home" pages refer to it. However, it is one of my tasks for this quarter to improve our vector docs, so I'll address this matter separately.

### CODE REVIEW

File: content/develop/connect/clients/go.md [hunk 1]
```diff
@@ -159,23 +219,283 @@ if err != nil {
...
+		DB:       0,
+		Protocol: 2,
+	})
+
+    // ...
+}
+```
+
+In this example, the user objects will be stored as hashes in the database. Use a `string`
+array in the form of name->value pairs to supply the data for the
```
Comment (dwdougherty) [inline content/develop/connect/clients/go.md:389]: Name-value or key-value are more consistent with our S.G.
```suggestion
array in the form of name-value pairs to supply the data for the
```

File: content/develop/connect/clients/go.md [hunk 2]
```diff
@@ -159,23 +219,283 @@ if err != nil {
...
+
+user3 := []string{
+    "name", "Paul Zamir",
+    "email", "paul.zamir@example.com",
+    "age", "35",
+    "city", "Tel Aviv",
+}
+```
+
+It is slightly simpler to create the index for hash objects than
```
Comment (dwdougherty) [inline content/develop/connect/clients/go.md:415]:
```suggestion
It is easier to create the index for hash objects than
```

**Review by dwdougherty — APPROVED** [review]
Just two advisory comments. I'll leave their application up to you.

**Comment by andy-stark-redis** [comment]
@dwdougherty Good suggestions, thanks :-)

**Pull request merged**